cms/djangoapps/contentstore/views/course.py

```diff
@@ -306,6 +306,101 @@ def course_rerun_handler(request, course_key_string):
             })
 
 
+def _update_children_in_module_store(xblock, user):
+    """
+    Recursively update all children of given xblock in modulestore
+    """
+    for child in xblock.get_children():
+        modulestore().update_item(child, user.id)
+        for grand_child in child.get_children():
+            _update_children_in_module_store(grand_child, user)
```

Comment:
You don't really need this additional loop. Just call ` _update_children_in_module_store(child, user)` instead.

Reply:
Additionally, I'm thinking since we need to loop through it all anyway, let's just remove the method from the block itself, since now we are traversing through this structure twice. 

Instead of setting it via `xblock.set_discussion_toggle(value)` just set it here when we're going through the entire structure.